### PR TITLE
[CI] Run SYCL-CTS with the latest nightly

### PR DESCRIPTION
**Pull request by KornevNikita**
Add a new workflow to run SYCL-CTS when new nightly build is available. This should help us to handle new failures in advance, monitor new suites and so on.

To start with running only reduced scope of tests (cts_exclude_filter is big enough) on unused "debug" runner (linux+cpu).

### CODE REVIEW

File: .github/workflows/sycl-linux-run-tests.yml [hunk 1]
```diff
@@ -81,6 +81,7 @@ on:
           - '["Linux", "gen12"]'
           - '["amdgpu"]'
           - '["Linux", "arc"]'
+          - '["debug"]'
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:84]: I can rename the label. How about `cts-cpu`?

File: .github/workflows/sycl-linux-run-tests.yml [hunk 2]
```diff
@@ -290,6 +291,12 @@ jobs:
...
+      # SIGSEGV while running test_all, so running each binary separately.
+      # BTW test_all requires a lot of resources to build it, so probably it'll
+      # be better to build each binary also separately.
+      # run: |
+      #   ./build-cts/bin/test_all --list-devices
+      #   ./build-cts/bin/test_all $CTS_TEST_ARGS
       run: |
-        ./build-cts/bin/test_all --list-devices
-        ./build-cts/bin/test_all $CTS_TEST_ARGS
+        for i in `ls -1 ./build-cts/bin`; do if [ "$i" != "test_all" ]; then echo "###### Running $i" ; build-cts/bin/$i; fi; done
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:302]: Please try doing
```
echo "::group::Running $i"
# run $i
echo "::endgroup::"
```

File: .github/workflows/sycl-linux-run-tests.yml [hunk 3]
```diff
@@ -290,6 +291,12 @@ jobs:
       if: inputs.tests_selector == 'cts'
       env:
         ONEAPI_DEVICE_SELECTOR: ${{ inputs.target_devices }}
+      # FIXME: For some reason the "sub_group api" test-case is failing with
+      # SIGSEGV while running test_all, so running each binary separately.
+      # BTW test_all requires a lot of resources to build it, so probably it'll
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:296]: Should we try per-source kernel splitting maybe?

File: .github/workflows/sycl-linux-run-tests.yml (hunk 1 above)
Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:84]: Sure, that would be more meaningful.

File: .github/workflows/sycl-linux-run-tests.yml (hunk 3 above)
Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:296]: I can't imagine other reasons for the slowdown...

File: .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    uses: ./.github/workflows/sycl-linux-run-tests.yml
+    with:
+      name: SYCL-CTS
+      runner: '["debug"]'
+      image: ghcr.io/intel/llvm/sycl_ubuntu2204_nightly:latest
+      image_options: -u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN
+      target_devices: opencl:cpu
+      tests_selector: cts
+      install_drivers: false
+      ref: ${{ github.sha }}
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml:20]: nit: isn't it the default?

File: .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  push:
+    tags:
+      - nightly-*
+
+jobs:
+  run:
+    uses: ./.github/workflows/sycl-linux-run-tests.yml
+    with:
+      name: SYCL-CTS
+      runner: '["debug"]'
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml:14]:
```suggestion
      runner: '["cts-cpu"]'
```

File: .github/workflows/sycl-linux-run-tests.yml [hunk 6]
```diff
@@ -290,6 +291,18 @@ jobs:
...
+      # be better to build each binary also separately.
+      # run: |
+      #   ./build-cts/bin/test_all --list-devices
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:299]: Can we have a separate step that will just "list-devices"?

File: .github/workflows/sycl-linux-run-tests.yml [hunk 7]
```diff
@@ -290,6 +291,18 @@ jobs:
...
+      #   ./build-cts/bin/test_all $CTS_TEST_ARGS
       run: |
-        ./build-cts/bin/test_all --list-devices
-        ./build-cts/bin/test_all $CTS_TEST_ARGS
+        for i in `ls -1 ./build-cts/bin`; do \
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:302]: Did it fail without `\`s?

File: .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+name: SYCL-CTS on Linux&CPU with the latest nightly
+run-name: SYCL-CTS - ${{ github.ref_name }}
+
+on:
+  push:
+    tags:
+      - nightly-*
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml:7]: Can we add manual workflow_dispatch as well? Possibly with a selectable runner/target devices.

File: .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml (hunk 4 above)
Comment (KornevNikita) [reply .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml:20]: Looks like no, see https://github.com/intel/llvm/actions/runs/8007137309

File: .github/workflows/sycl-linux-run-tests.yml (hunk 2 above)
Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: [918399f](https://github.com/intel/llvm/pull/12811/commits/918399f035fab9830a7358c553470a67baaf0789)

File: .github/workflows/sycl-linux-run-tests.yml (hunk 6 above)
Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml:299]: Do you mean exactly `./build-cts/bin/test_all --list-devices`, or just `sycl-ls`?

Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:299]: cts's "--list-devices". I believe `sycl-ls` is part of `sycl-linux-run-tests.yml` already.

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: Like this [cd94539](https://github.com/intel/llvm/pull/12811/commits/cd9453996255e0e4dc8eb0f675a13de7422a53a9)?

File: .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml (hunk 8 above)
Comment (KornevNikita) [reply .github/workflows/sycl-cts-linux-cpu-latest-nightly.yml:7]: As discussed offline we can just run `sycl-linux-run-tests.yml` as it has `workflow_dispatch`.

File: .github/workflows/sycl-linux-run-tests.yml (hunk 7 above)
Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: Yep, works without.

File: .github/workflows/sycl-linux-run-tests.yml (hunk 6 above)
Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml]: Does it exit immediately after that without running tests? If so, then yes!

File: .github/workflows/sycl-linux-run-tests.yml [hunk 9]
```diff
@@ -286,10 +287,27 @@ jobs:
         -DDPCPP_INSTALL_DIR="$$(dirname (which clang++))/.." \
         $CMAKE_EXTRA_ARGS
         ninja -C build-cts
+
+    - name: List devices
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:291]:
```suggestion
    - name: [CTS] List devices
```

Comment (KornevNikita) [reply .github/workflows/sycl-linux-run-tests.yml]: [c97e9dc](https://github.com/intel/llvm/pull/12811/commits/c97e9dcf4e9dc804fc9b42e74542cd7dba951037)

**Review by aelovikov-intel — APPROVED** [review]

File: .github/workflows/sycl-nightly.yml [hunk 10]
```diff
@@ -107,6 +107,22 @@ jobs:
       runner: '["Windows","gen12"]'
       sycl_toolchain_archive: ${{ needs.build-win.outputs.artifact_archive_name }}
 
+  sycl-cts:
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-nightly.yml:110]: I think we can actually merge that into the e2e matrix.

File: .github/workflows/sycl-linux-run-tests.yml [hunk 11]
```diff
@@ -286,10 +287,27 @@ jobs:
         -DDPCPP_INSTALL_DIR="$$(dirname (which clang++))/.." \
         $CMAKE_EXTRA_ARGS
         ninja -C build-cts
+
+    - name: [CTS] List devices
```
Comment (aelovikov-intel) [inline .github/workflows/sycl-linux-run-tests.yml:291]: nit:
```suggestion
    - name: SYCL CTS List devices
```

Sorry, I didn't expand the context before.

File: .github/workflows/sycl-nightly.yml (hunk 10 above)
Comment (KornevNikita) [reply .github/workflows/sycl-nightly.yml]: [441647a](https://github.com/intel/llvm/pull/12811/commits/441647aa10085f0ba6b4be8eb93359f7f844da05)
Not sure what will be `matrix.tests_selector` for the rest

**Comment by aelovikov-intel** [comment]
Being tested in https://github.com/intel/llvm/actions/runs/8024711118.

**Review by aelovikov-intel — APPROVED** [review]

File: .github/workflows/sycl-linux-run-tests.yml (hunk 3 above)
Comment (AlexeySachkov) [reply .github/workflows/sycl-linux-run-tests.yml:302]: `per-source` splitting is a default mode of splitting

**Pull request merged**